### PR TITLE
libvpx: update to 1.8.0

### PR DESCRIPTION
**Pull request by q66**

### CODE REVIEW

**Comment by q66** [comment]
This will also make it possible to update Qt5 to 5.12.x. I will do this separately first though because otherwise there would be a cycle.

**Comment by maxice8** [comment]
I'd recommend making libvpx a versioned package so we can have 2 of it installed at the same time to make transitions easier

**Comment by q66** [comment]
There won't be any need to, as far as I can see it's fully backwards compatible, but there is an ABI bump because of added fields.

**Comment by q66** [comment]
Though, other distros do version it, so maybe a good idea to do so.

**Comment by maxice8** [comment]
It is more about easing transitions for us

File: srcpkgs/libvpx/template [hunk 1]
```diff
@@ -1,14 +1,14 @@
 # Template file for 'libvpx'
 pkgname=libvpx
-version=1.7.0
+version=1.8.0
 revision=1
 hostmakedepends="perl yasm"
 short_desc="The VP8/VP9 Codecs"
```
Comment (maxice8) [inline srcpkgs/libvpx/template]: please remove the article

**Comment by Cogitri** [comment]
We might be able to use system libvpx for chromium with this, will test in a bit.

**Comment by q66** [comment]
Looks like gst-plugins-good1 fails, needs fix

```
gstvpxdec.c: In function 'gst_vpx_dec_post_processing_flags_get_type':
gstvpxdec.c:65:15: error: 'VP8_DEBUG_TXT_FRAME_INFO' undeclared (first use in this function); did you mean 'VP9D_GET_FRAME_SIZE'?
     {C_FLAGS (VP8_DEBUG_TXT_FRAME_INFO),
```

**Comment by q66** [comment]
I have a patch now. Seems like it should be the only one necessary.

**Comment by q66** [comment]
Firefox also fails:

```
 3:47.83 /builddir/firefox-65.0/media/webrtc/trunk/webrtc/modules/video_coding/codecs/vp9/vp9_impl.cc:821:17: error: 'struct vpx_svc_ref_frame_config' has no member named 'frame_flags'
 3:47.83          sf_conf.frame_flags[layer_idx] = layer_flags;
```

**Comment by q66** [comment]
Looks like patching Firefox (or rather, the webrtc component) will be non-trivial. Should we switch to vendor libvpx for the time being?

**Comment by Cogitri** [comment]
I'd opt for keeping libvpx1.7

**Comment by q66** [comment]
So, introduce libvpx5 + libvpx5-devel, use that?

**Comment by Cogitri** [comment]
Yup.

**Comment by q66** [comment]
Sounds good. Building qt5 now, let's see if that one will also need patching. If yes, I will switch to the older one for the time being as well, and switch over to the more recent one when we update Qt.

**Comment by q66** [comment]
It seems this will be problematic since it installs `/usr/include/vpx` in both cases as well as `vpx.pc` and `libvpx.a`/`libvpx.so`. Any suggestions on how to deal with conflicts?

**Comment by Cogitri** [comment]
Hm, that's a problem indeed. I guess we _could_ have the devel packages conflict with each other since the non-devel ones won't conflict, right?

**Comment by q66** [comment]
Yes, that's right

**Comment by q66** [comment]
So, i guess I will add `conflicts="libvpx6-devel>=0"` into `libvpx5-devel` and we'll have to live with those two conflicting while there's still stuff depending on it.

**Comment by Cogitri** [comment]
Sounds good to me.

**Comment by q66** [comment]
Okay, looks like we're good generally. Can someone test building `avidemux` and `virtualbox-ose`? Those are the only two I can't test.

**Comment by Hoshpak** [comment]
Avidemux builds but does not link against libvpx, neither does the version currently in the repository. virtualbox-ose fails to detect any library, not just libvpx but it already does this in the master branch so this is a separate issue. I don't have time to investigate right not but will create an issue.

**Comment by Hoshpak** [comment]
#8418

**Comment by q66** [comment]
Okay then, what do you suggest?

**Comment by Hoshpak** [comment]
Either wait for it to be fixed or mark the virtualbox-ose package as broken for now. I suspect that the issue is not caused by the package itself but by some package in the toolchain that broke with a recent update since virtualbox-ose built correctly after being updated. However I probably won't get around to investigating this before next week.

**Comment by q66** [comment]
What about avidemux? I wasn't completely sure what you meant there; is it broken even without libvpx update? Or does it simply mean I have to link it against libvpx5 for now?

**Comment by Cogitri** [comment]
```
➜ xbps-query -R avidemux
architecture: x86_64
build-date: 2019-01-25 17:22 UTC
filename-sha256: 52248a80eec6760e1079107ad0b4d22c5adc5841b2397100d3d563198444cbad
filename-size: 6987KB
homepage: http://avidemux.sourceforge.net/
installed_size: 26MB
license: GPL-2.0-or-later
maintainer: Jürgen Buchmüller <pullmoll@t-online.de>
pkgver: avidemux-2.7.1_6
repository: https://alpha.de.repo.voidlinux.org/current
shlib-provides:
	libADM_coreSocket6.so
	libADM_coreImage6.so
	libADM_coreVideoFilter6.so
	libADM_UI_Cli6.so
	libADM_coreJobs.so
	libADM_coreScript.so
	libADM_render6_cli.so
	libADM_coreVDPAU6.so
	libADM_coreMuxer6.so
	libADM_coreLibVA6.so
	libADM_coreUI6.so
	libADM_coreAudioEncoder6.so
	libADM_coreAudio6.so
	libADM_coreLibVAEnc6.so
	libADM_coreAudioDevice6.so
	libADM_coreImageLoader6.so
	libADM_coreDemuxerMpeg6.so
	libADM6avutil.so.55
	libADM6postproc.so.54
	libADM_core6.so
	libADM6swscale.so.4
	libADM_UIQT56.so
	libADM_render6_QT5.so
	libADM_coreUtils6.so
	libADM_coreSubtitle.so
	libADM_audioParser6.so
	libADM_openGLQT56.so
	libADM_coreVideoCodec6.so
	libADM6avcodec.so.57
	libADM6avformat.so.57
	libADM_coreSqlLight3.so
	libADM_coreAudioFilterAPI6.so
	libADM_coreVideoEncoder6.so
	libADM_coreDemuxer6.so
shlib-requires:
	libpthread.so.0
	libstdc++.so.6
	libm.so.6
	libgcc_s.so.1
	libc.so.6
	libQt5Network.so.5
	libX11.so.6
	libQt5OpenGL.so.5
	libQt5Widgets.so.5
	libQt5Gui.so.5
	libGL.so.1
	libQt5Core.so.5
	libmp3lame.so.0
	libvorbis.so.0
	libvorbisenc.so.2
	libfaac.so.0
	libmp4v2.so.2
	libva.so.2
	libx265.so.169
	libxvidcore.so.4
	libx264.so.152
	libass.so.9
	liba52.so.0
	libmad.so.0
	libdca.so.0
	libfaad.so.2
	libasound.so.2
	libjack.so.0
	libva-x11.so.2
	libdl.so.2
	libvdpau.so.1
	libva-drm.so.2
	libXv.so.1
	libGLU.so.1
	libz.so.1
	libsqlite3.so.0
short_desc: Video editing and processing application
source-revisions: avidemux:d7e38d5797
```

Doesn't seem to me that avidemux links against libvpx.

**Comment by q66** [comment]
Good. Could you test if it builds with libvpx removed from makedepends?

**Comment by q66** [comment]
Updated for those two.

**Comment by Cogitri** [comment]
> Good. Could you test if it builds with libvpx removed from makedepends?

avidemux builds fine for me without the dep

**Comment by Hoshpak** [comment]
It seems I either had an outdated masterdir or the issue was fixed in the meantime. I can confirm that virtualbox-ose builds fine with and links against libvpx6 so you can remove the broken tag.

**Comment by q66** [comment]
Done.

**Comment by Hoshpak** [comment]
Is this good to merge or is there anything else to do?

**Comment by q66** [comment]
I think that's all

**Comment by q66** [comment]
Cool. Looks like everything but browser cores are ok with new libvpx then. I'll update Qt5 to 5.12.x next, now that it's possible.

**Pull request merged**